Clarify user session limits documentation and test SSO scenario (#19372)
keycloak/keycloak: docs/documentation/server_admin/topics/authentication/flows.adoc

Note that the user session limits should be added to your bound *Browser flow*, *Direct grant flow*, *Reset credentials* and also to any *Post broker login flow*.
The authenticator should be added at the point when the user is already known during authentication (usually at the end of the authentication flow) and should be typically REQUIRED. Note that it is not possible to have
ALTERNATIVE and REQUIRED executions at the same level.

For most of authenticators like `Direct grant flow`, `Reset credentials` or `Post broker login flow`, it is recommended to add the authenticator as REQUIRED at the end of the authentication flow.

> For example it can look like this for `Reset credentials` flow:

Reviewer: ```suggestion
Here is an example for the `Reset credentials` flow:
```

image:images/authentication-user-session-limits-resetcred.png[Authentication User Session Limits Reset Credentials Flow]

> For `Browser` flow, it can be beneficial to not add the Session Limits authenticator at the top level flow. This is due the `Cookie` authenticator, which automatically re-authenticates users based
> on SSO cookie, is at top level and we do not want to check session limits during SSO re-authentication as user session already exists. So instead, it can be good to add separate ALTERNATIVE

Reviewer: ```suggestion
For `Browser` flow, consider not adding the Session Limits authenticator at the top level flow. This recommendation is due to the `Cookie` authenticator, which automatically re-authenticates users based
on SSO cookie. It is at the top level and it is better to not check session limits during SSO re-authentication because a user session already exists. So instead, consider adding a separate ALTERNATIVE
```

> subflow (like `authenticate-user-with-session-limit` at the example below) at the same level like `Cookie`. Then it it possible to add REQUIRED subflow (in the example below
> `real-authentication-subflow`) as nested subflow of `authenticate-user-with-session-limit` and add `User Session Limit` at the same level as well. Inside `real-authentication-subflow`,

Reviewer: ```suggestion
subflow, such as the following `authenticate-user-with-session-limit` example) at the same level like `Cookie`. Then you can add a REQUIRED subflow (in the following `real-authentication-subflow`example) as a nested subflow of `authenticate-user-with-session-limit` and add a `User Session Limit` at the same level as well. Inside the `real-authentication-subflow`,
```

> you can add real authenticators similarly like in the default browser flow. For example you can see the example flow below, which allows to authenticate users either with identity provider or

Reviewer: ```suggestion
you can add real authenticators in a similar fashion to the default browser flow. The following example flow allows to users to authenticate with and identity provider or
```

with password and OTP:

image:images/authentication-user-session-limits-browser.png[Authentication User Session Limits Browser Flow]

> Regarding `Post Broker login flow`, it can be fine to just add the `User Session Limits` as only authenticator in the authentication flow (as long as you do not have any other authenticators,
> which you want to trigger after authentication with your identity provider) and make sure that this flow is configured as `Post Broker Flow` at your identity providers. This is needed, so

Reviewer: ```suggestion
Regarding `Post Broker login flow`,  you can add the `User Session Limits` as the only authenticator in the authentication flow  as long as you have no other authenticators that you want to trigger after authentication with your identity provider.  However, make sure that this flow is configured as `Post Broker Flow` at your identity providers. This requirement exists needed so that 
```

the authentication with Identity providers also participates in the session limits.

> NOTE: Currently, the administrator is responsible for maintaining consistency between the different configurations. So make sure that all your flows use same configuration

Reviewer: ```suggestion
NOTE: Currently, the administrator is responsible for maintaining consistency between the different configurations. So make sure that all your flows use same the configuration
```
